Added ability to add message_properties with variables.

sophie135 opened the pull request:
When adding message_properties dictionary (message_properties => {"content_type" => "const type"})
You can now use dynamic values from variables {"content_type" => "%{ContentType}"}

Thanks for contributing to Logstash! If you haven't already signed our CLA, here's a handy link: https://www.elastic.co/contributor-agreement/

jsvd commented on lib/logstash/outputs/rabbitmq.rb:
```diff
@@ -63,7 +63,7 @@ def multi_receive_encoded(events_and_data)
       def publish(event, message)
         raise ArgumentError, "No exchange set in HareInfo!!!" unless @hare_info.exchange
-        local_exchange.publish(message, :routing_key => event.sprintf(@key), :properties => symbolize(@message_properties.merge(:persistent => @persistent)))
+        local_exchange.publish(message, :routing_key => event.sprintf(@key), :properties => symbolize(@message_properties.merge(:persistent => @persistent), event))
```
can we revert this change, I don't see a need to log the whole event

sophie135 replied:
I didn't understand what you meant.
I tried to evaluate the value in the {key: value}, so when it's a variable, like {"reply_to" => "%{myReplyTo}"}, the value in rabbit won't be "%{myReplyTo}" but the value of the variable myReplyTo.
I know there is an issue with the commit now (and the reason it didn't passed the tests), that it refers to the value always as a string, and if it's a boolean (like {"persistent" => true}) it will return  {"persistent" => "true"}.
I'll fix it. But is there any other things I'm missing?

sophie135 replied:
I didn't understand what you meant.
I tried to evaluate the value in the {key: value}, so when it's a variable, like {"reply_to" => "%{myReplyTo}"}, the value in rabbit won't be "%{myReplyTo}" but the value of the variable myReplyTo.
I know there is an issue with the commit now (and the reason it didn't passed the tests), that it refers to the value always as a string, and if it's a boolean (like {"persistent" => true}) it will return  {"persistent" => "true"}.
I'll fix it. But is there any other things I'm missing?

sophie135 committed:
Added ability to add message_properties with variables.

When adding message_properties dictionary (message_properties => {"content_type" => "const type"})
You can now use dynamic values from variables {"content_type" => "%{ContentType}"}

jeslopalo commented:
hi @jsvd @sophie135, 

is there any plan to release this feature?

It would be very useful to be able to set the priority of the message dynamically.

`message_properties => {"priority" => "%{priority}"}`

yaauie commented:
@jeslopalo I have revived this feature in the new RabbitMQ Integration Plugin: https://github.com/logstash-plugins/logstash-integration-rabbitmq/pull/9

Since the Integration Plugin requires Logstash 6.5, I'd appreciate your input on backport requirements to get this feature working for you since you contributed.
